driver-reactive-streams/src/main/com/mongodb/reactivestreams/client/internal/ClientSessionPublisherImpl.java

```diff
@@ -186,9 +193,11 @@ public AsyncClientSession getWrapped() {
                     new AbortTransactionOperation(transactionOptions.getWriteConcern())
                             .recoveryToken(getRecoveryToken()),
                     readConcern, this)
-                    .doOnError(MongoException.class, this::unpinServerAddressOnError)
                     .onErrorResume(Throwable.class, (e) -> Mono.empty())
-                    .doOnTerminate(() ->  cleanupTransaction(TransactionState.ABORTED));
+                    .doOnTerminate(() -> {
+                        setPinnedServerAddress(null);
```

Comment:
The spec requires that abort unpin on success as well as error, but there were no tests for it.  Now there are, and Java driver fails the test.  This fixes it for reactive